### PR TITLE
Training/2021.11 E1 - Advanced Tasking

### PR DESCRIPTION
**Pull request by leogermond**
Moves the rst source code to executable Ada files that are extracted and reinjected onto the course. This allowed for the correction of some bugs, and provides fully running examples to trainees for complex cases.

### CODE REVIEW

File: courses/fundamentals_of_ada/adv_240_tasking.rst [hunk 1]
```diff
@@ -334,35 +413,58 @@ What of the following completions for :ada:`P`'s members is illegal?
...
+======
+
+-------------
+Delay keyword
+-------------
+
+- :ada:`delay` keyword part of tasking
+- Blocks for a time
+- Relative: Blocks for at least :ada:`Duration`
+- Absolute: Blocks until a given :ada:`Calendar.Time` or :ada:`Real_Time.Time`
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_240_tasking.rst:427]: until no earlier than

File: courses/fundamentals_of_ada/adv_240_tasking.rst [hunk 2]
```diff
@@ -334,35 +413,58 @@ What of the following completions for :ada:`P`'s members is illegal?
...
+      V2 := new First_T;  -- V2 is activated immediately
 
-* Tasks instantiated on the heap are activated right away
-* Tasks are limited objects (no copies allowed)
+--------------------
+Single Declaration
+--------------------
 
-.. code:: Ada
+ * Instanciate an **anonymous** task (or protected) type
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_240_tasking.rst:466]: Instantiate

**Review by frank-at-adacore — CHANGES_REQUESTED** [review]

File: courses/fundamentals_of_ada/adv_240_tasking.rst (hunk 1 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_240_tasking.rst:427]: Done

File: courses/fundamentals_of_ada/adv_240_tasking.rst (hunk 2 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_240_tasking.rst:466]: Done

**Review by frank-at-adacore — APPROVED** [review]

**Pull request merged**